### ISSUE TITLE
Making sure I'm understanding the rules correctly...

### ISSUE THREAD

**Issue by binaryfire** [chaifeng/ufw-docker#77]
Hey @chaifeng

Thanks for a great solution! I'd like to customise the rules for my needs and want to make sure I've understood them correctly:

* Rules ending in 192.168.0.0/16 allow all published ports to be accessed from 192.168.x.x addresses
* Rules ending in 10.0.0.0/8 allow all published ports to be accessed from 10.x.x.x addresses

Is that right? 

I'm going for a zero-trust network so I want to drop all traffic by default (including from private network addresses). And then add UFW rules for the specific private and public ips I want to allow. 

**So was thinking of these rules:**

```
# BEGIN UFW AND DOCKER
*filter
:ufw-user-forward - [0:0]
:ufw-docker-logging-deny - [0:0]
:DOCKER-USER - [0:0]
-A DOCKER-USER -j ufw-user-forward

-A DOCKER-USER -p udp -m udp --sport 53 --dport 1024:65535 -j RETURN

-A DOCKER-USER -j ufw-docker-logging-deny -p tcp -m tcp --tcp-flags FIN,SYN,RST,ACK SYN -d 172.16.0.0/12
-A DOCKER-USER -j ufw-docker-logging-deny -p udp -m udp --dport 0:32767 -d 172.16.0.0/12

-A DOCKER-USER -j RETURN

-A ufw-docker-logging-deny -m limit --limit 3/min --limit-burst 10 -j LOG --log-prefix "[UFW DOCKER BLOCK] "
-A ufw-docker-logging-deny -j DROP

COMMIT
# END UFW AND DOCKER
```

**Combined with UFW commands like this:**

* Allow private network access to published ports 9000 and 22:
`ufw route allow proto tcp from 192.168.0.0/16 to any port 9000,22`

* Allow public IP 5.5.5.5 access to published ports 5000:
`ufw route allow proto tcp from 5.5.5.5 to any port 5000`

Is this the right approach? Or am I missing something?

Thanks!

**Comment by binaryfire** [comment]
@chaifeng EDIT: I also don't need to log dropped requests, so I changed ufw-docker-deny to ufw-docker-logging-deny to ufw-docker-deny and removed the logging entry

**Comment by chaifeng** [comment]
> 

Yes, just remove the three lines of code about `192.168.0.0/16` if you don't trust this network.

**Comment by chaifeng** [comment]
> 



> @chaifeng Also, I don't need to log dropped requests.
> 
> Will `ufw logging off` also switch off logging for ufw-docker-logging-deny? Or do I need to remove `-A ufw-docker-logging-deny -m limit --limit 3/min --limit-burst 10 -j LOG --log-prefix "[UFW DOCKER BLOCK] "` ?
> 
> Thanks :)

Correct!

**Comment by binaryfire** [comment]
@chaifeng Thank you!

Just to confirm- are the 10.0.0.0/8 and 172.16.0.0/12 ranges for Docker’s internal networking?

**Comment by chaifeng** [comment]
> @chaifeng Thank you!
> 
> Just to confirm- are the 10.0.0.0/8 and 172.16.0.0/12 ranges both required for Docker’s internal networking?

depends on your configurations of docker/swarm networks, for example, https://docs.docker.com/network/bridge/#configure-the-default-bridge-network
